### PR TITLE
[PUBDEV-3537] fix model.start()

### PR DESCRIPTION
**Pull request by st-pasha**
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/438)

### CODE REVIEW

**Comment by abal5** [comment]
@h2o-ops please test

File: h2o-py/h2o/estimators/estimator_base.py [hunk 1]
```diff
@@ -184,53 +187,43 @@ def train(self, x=None, y=None, training_frame=None, offset_column=None, fold_co
                     if xi not in names:
                         raise H2OValueError("Column %s not in the training frame" % xi)
                     xset.add(xi)
+        x = list(xset)
 
-        parms["x"] = list(xset)
```
Comment (mmalohlava) [inline h2o-py/h2o/estimators/estimator_base.py]: Is `x` used at all at the end?

Comment (st-pasha) [inline h2o-py/h2o/estimators/estimator_base.py]: Yes, in line 214 to construct `ignored_columns` which is then passed along to the server.

Although it does seem there's lots of cruft still in the code... for example `x` never has to be transformed from set to a list

**Comment by st-pasha** [comment]
In my private jenkins only 3 jobs failed on this PR: R_cmd_check_as_Cran, runit_small_client_mode, and win7_pyunit_small. In the latter there was only 1 failed test with "ValueError: File not found: smalldata/jira/test_pubdev3589.txt". 

So I'd say it works as intended.

**Pull request merged**